### ISSUE TITLE
Difference b/w Matrix multiplication and Matrix Dot Product

### ISSUE THREAD

**Issue by haiderabbas007** [ubsuny/CP1-24-HW2#38]
What's the difference b/w Matrix multiplication and Matrix Dot Product?

**Comment by p-pxmpo** [comment]
from my understanding the dot product is a way to multiply two vectors

**Comment by p-pxmpo** [comment]
Say we have two vectors a = <a1, a2, a3> and b = <b1, b2, b3> then the dot product a dot b = (a1b1) + (a2b2) + (a3b3)

**Comment by p-pxmpo** [comment]
The output is a scalar not a vector but thats not relevant to this project

**Comment by p-pxmpo** [comment]
could somebody verify if this is correct please?

**Comment by haiderabbas007** [comment]
Yes, but this code returns a scalar (=42), whereas matrices are involved instead of vectors.
![image](https://github.com/user-attachments/assets/37e6fc60-c697-4bfb-88ab-ea747bc1112f)

**Comment by ojha-aditya** [comment]
Can you share it in a code format? We could try and run it that way

**Comment by ojha-aditya** [comment]
Regardless, the major difference between the two is simply in the definition as given above, the Matrix Dot Product is like treating matrices as elements of a vector space, while the conventional matrix multiplication simply takes the matrix product.

The former is like defining a map that takes two different matrices to their 'inner product'

**Comment by HeshamElsaman** [comment]
I agree with that. Adding my understanding, the inner product is a map from the vector space to scalars (which might be complex in general) that satisfies some criteria. It's just that the inner product (dot product in this case) is defined differently for those are two different vector spaces (vectors and matrices).

**Comment by ojha-aditya** [comment]
Yes. And in general numpy.dot() function implements the dot product in special cases, for matrices it doesn't automatically give out the dot product as defined in the code block shared above. 

Check here: https://numpy.org/doc/stable/reference/generated/numpy.dot.html

**Comment by haiderabbas007** [comment]
Thanks for the clarification maths geeks :)
